pkg/providers/ingress/ingress.go

```diff
@@ -131,7 +130,7 @@ func (c *ingressController) sync(ctx context.Context, ev *types.Event) error {
 	}
 
 	if err != nil {
-		if !k8serrors.IsNotFound(err) {
+		if !k8serrors.IsNotFound(err) || (k8serrors.IsNotFound(err) && ev.Type != types.EventDelete) {
```

Comment:
There are two questions here
1. Events are not always up-to-date. We always retrieve the latest object, which may have been deleted. This logic can lead to infinite retries.
2. Redundant logic, same as line 138.

![image](https://user-images.githubusercontent.com/79972061/231963858-9486017a-acec-4b9c-afc5-06b3ee747a1e.png)

```go
if !k8serrors.IsNotFound(err) || (k8serrors.IsNotFound(err) && ev.Type != types.EventDelete) { return }
if ev.Type != types.EventDelete { return } # Redundant 

<==>

if !k8serrors.IsNotFound(err) { return }
if ev.Type != types.EventDelete { return }

```